Source/RpmAvatarCreator/Private/UI/RpmAvatarEditorUI.cpp

```diff
@@ -200,3 +257,72 @@ void URpmAvatarEditorUI::SetColorSelectedPin(ERpmPartnerAssetColor AssetColor, i
 		ColorButton->SetSelected(IsSelected);
 	}
 }
+
+void URpmAvatarEditorUI::DownloadAssetIcons(const UWrapBox* AssetContainer)
+{
+	for (UWidget* Widget : AssetContainer->GetAllChildren())
+	{
+		URpmAssetButtonUI* AssetButton = Cast<URpmAssetButtonUI>(Widget);
+		const FString& IconUrl = AssetButton->Asset.IconUrl;
+		if (!IconUrl.IsEmpty())
+		{
+			AssetImageMap.Add(IconUrl, AssetButton);
+			AvatarCreatorApi->DownloadImage(IconUrl, 64, AssetIconDownloadCompleted);
+		}
+	}
+}
+
+void URpmAvatarEditorUI::OnAssetIconDownloaded(UTexture2D* Image, const FString& Url)
+{
+	if (AssetImageMap.Contains(Url))
```

Comment:
this if statement is the same logic as the the If statement in OnTemplateIconDownloaded, it's rather minor but perhaps there is a way to remove some of the logic/code duplication?